AudioManagerUtilities/src/CAmCommonAPIWrapper.cpp

```diff
@@ -207,20 +207,37 @@ void CAmCommonAPIWrapper::registerTimeout(CommonAPI::Timeout* timeout, const Com
 {
     timespec pollTimeout;
     int64_t localTimeout = timeout->getTimeoutInterval();
-
-    pollTimeout.tv_sec = localTimeout / 1000;
-    pollTimeout.tv_nsec = (localTimeout % 1000) * 1000000;
+    
+    if(CommonAPI::TIMEOUT_INFINITE==localTimeout)//dispatch never
+    {
+        pollTimeout.tv_sec = localTimeout;
+        pollTimeout.tv_nsec = 0;
+    }
+    else if(CommonAPI::TIMEOUT_NONE==localTimeout)//dispatch immediately
+    {
+        pollTimeout.tv_sec = 0;
+        pollTimeout.tv_nsec = 1000000;     
+    }
+    else 
+    {
+        pollTimeout.tv_sec = localTimeout / 1000;
+        pollTimeout.tv_nsec = (localTimeout % 1000) * 1000000;   
+    }
 
     //prepare handle and callback. new is eval, but there is no other choice because we need the pointer!
     sh_timerHandle_t handle;
 
     //add the timer to the pollLoop
-    mpSocketHandler->addTimer(pollTimeout, &pCommonTimerCallback, handle, timeout);
-
-    timerHandles myHandle({handle,timeout});
-    mpListTimerhandles.push_back(myHandle);
-
-    return;
+    am_Error_e error = mpSocketHandler->addTimer(pollTimeout, &pCommonTimerCallback, handle, timeout);
+    if (error != am_Error_e::E_OK || handle == 0)
```

Comment:
Probably OK - just a reflection.  Long term it might be good to be more strict about what is the actual error condition?  Is it the return value or that handle is zero?  Can one or the other happen, or do they always happen together?  The way it is written now it looks almost like we're uncertain if the function always does the right thing with the return/error value...  and being uncertain is probably a bad sign.  

If they have a relationship it might useful to assert() this instead.  E.g. if error, assert that handle is always expected to be zero.